### ISSUE TITLE
puppetlabs-inifile and puppet-samba cannot coexist

### ISSUE THREAD

**Issue by bitte-ein-bit** [kakwa/puppet-samba#9]
puppet-samba shadows the original IniFile ruby-classes. Therefore an error shows up. 

```
Error: /Stage[main]/Profiles::Basic::Puppet/Ini_setting[templatedir]: Could not evaluate: wrong number of arguments (4 for 2)
/var/lib/puppet/lib/puppet/util/smb_file.rb:8:in `initialize'
/var/lib/puppet/lib/puppet/provider/ini_setting/ruby.rb:115:in `new'
/var/lib/puppet/lib/puppet/provider/ini_setting/ruby.rb:115:in `ini_file'
/var/lib/puppet/lib/puppet/provider/ini_setting/ruby.rb:41:in `exists?'
```

Solutions are either going back to puppetlabs-inifile and reverting 130f8068bdc656281e3b219fde8df48d9af3aacf or doing a real fork of inifile and rename the puppet util ruby-classes to something unique/specific to samba.

**Comment by kakwa** [comment]
Sorry, I taught I've forked it completely.

It should be fixed by https://github.com/kakwa/puppet-samba/commit/ae578e1a8a09b18d7c97a9a3c3e17599a1545952

This fix will come with version 0.3.1 of this module.

For reference, I've made the choice of forking it because at the time, the puppetlabs-inifile module was buggy.
What's more, it didn't removed ini sections when purged, which is problematic since sections in samba conf file define shares.
Maybe one day, when I find the time, I will take a closer look at puppetlabs recent modifications.
- https://tickets.puppetlabs.com/browse/MODULES-1998
- https://github.com/kakwa/puppet-samba/blob/master/manifests/dc.pp#L381
